### ISSUE TITLE
Ensure Keyboard Accessibility for Interactive Elements

### ISSUE THREAD

**Issue by Anastasiia-Bakanova** [NataGildry/team-challenge-online-shop#50]
Users who rely on keyboards for navigation require that all interactive elements are fully operable without a mouse.

Requirements:
Ensure all interactive elements (e.g., buttons, links, inputs) are operable using the keyboard.
Use a logical and predictable tab order to ensure smooth navigation.
Provide a visible focus indicator (e.g., outline or underline) for all focusable elements.

Verify the following:
Buttons respond to both Enter and Space keys.
Links are activated using the Enter key.

Acceptance Criteria:
All buttons can be activated using Enter and Space.
Links are activated with the Enter key.
Tab navigation follows a logical and predictable order.
All focusable elements have visible focus indicators during navigation.

Technical Notes:
Validate implementation with keyboard testing and accessibility tools like Axe.

**Comment by AlexnderDoroshenko** [comment]
1)
Info about issue:
https://dequeuniversity.com/rules/axe/4.10/button-name?application=AxeChrome

Elements with issue:

Element Location:
shared-small-card:nth-child(2) > .h-\[375px\].pt-6.flex-col > .pt-4.justify-end.pr-5 > a > shared-icon-button > .w-\[45px\].h-\[45px\].tracking-wider2
Element Location:
shared-small-card:nth-child(3) > .h-\[375px\].pt-6.flex-col > .pt-4.justify-end.pr-5 > a > shared-icon-button > .w-\[45px\].h-\[45px\].tracking-wider2
Element Location:
shared-small-card:nth-child(4) > .h-\[375px\].pt-6.flex-col > .pt-4.justify-end.pr-5 > a > shared-icon-button > .w-\[45px\].h-\[45px\].tracking-wider2

<button class="uppercase flex w-[45px] h-[45px] text-white rounded-full bg-black font-[15px] tracking-wider2 items-center justify-center hover:shadow-btnBlack active:opacity-50 active:shadow-none disabled:opacity-50 disabled:hover:shadow-none">

autoetests result:
![Image](https://github.com/user-attachments/assets/8bee3a7f-253c-465f-8e3c-3561cd9fa129)

2)
Info about issue:
https://dequeuniversity.com/rules/axe/4.10/link-name?application=AxeChrome

Elements with issue:

Element Location:
shared-small-card:nth-child(2) > .h-\[375px\].pt-6.flex-col > .pt-4.justify-end.pr-5 > a
<a href="/team-challenge-online-shop/catalog/product/1">

Element Location:
shared-small-card:nth-child(3) > .h-\[375px\].pt-6.flex-col > .pt-4.justify-end.pr-5 > a
<a href="/team-challenge-online-shop/catalog/product/2">

Element Location:
shared-small-card:nth-child(4) > .h-\[375px\].pt-6.flex-col > .pt-4.justify-end.pr-5 > a
<a href="/team-challenge-online-shop/catalog/product/3">

Element Location:
a[href$="instagram.com/"]
<a href="https://www.instagram.com/" target="_blank">

Element Location:
a[href$="web.telegram.org/"]
<a href="https://web.telegram.org/" target="_blank">

Element Location:
a[href$="twitter.com/"]
<a href="https://twitter.com/" target="_blank">

autoetests result:
![Image](https://github.com/user-attachments/assets/0a2d70fa-96bd-4217-9a77-e40a21bbb4e4)
